src/Bridges/lazy_bridge_optimizer.jl

```diff
@@ -402,79 +272,92 @@ end
+function bridge_index(b::LazyBridgeOptimizer, F::Type{<:MOI.AbstractFunction})
+    return bridge_index(b.graph, node(b, F))
 end
 
 supports_constraint_bridges(::LazyBridgeOptimizer) = true
+function supports_bridging_constrained_variable(
+    b::LazyBridgeOptimizer, ::Type{MOI.Reals}
+)
+    return b.variable_free_bridge_type !== nothing
+end
 function supports_bridging_constrained_variable(
     b::LazyBridgeOptimizer, S::Type{<:MOI.AbstractSet}
 )
-    update!(b, (S,))
-    return (S,) in keys(b.variable_best)
+    return !iszero(bridge_index(b, S))
 end
 function supports_bridging_constraint(
     b::LazyBridgeOptimizer, F::Type{<:MOI.AbstractFunction},
     S::Type{<:MOI.AbstractSet}
 )
-    update!(b, (F, S))
-    return (F, S) in keys(b.constraint_best)
+    return !iszero(bridge_index(b, F, S))
 end
 function supports_bridging_objective_function(
     b::LazyBridgeOptimizer, F::Type{<:MOI.AbstractScalarFunction}
 )
-    update!(b, (F,))
-    return (F,) in keys(b.objective_best)
+    return !iszero(bridge_index(b, F))
+end
+
+function bridge_type(b::LazyBridgeOptimizer, ::Type{MOI.Reals})
+    if b.variable_free_bridge_type === nothing
+        throw(MOI.UnsupportedConstraint{MOI.VectorOfVariables, MOI.Reals}())
+    end
+    return b.variable_free_bridge_type
 end
 function bridge_type(b::LazyBridgeOptimizer, S::Type{<:MOI.AbstractSet})
-    update!(b, (S,))
-    result = get(b.variable_best, (S,), nothing)
-    if result === nothing
+    index = bridge_index(b, S)
+    if iszero(index)
         throw(MOI.UnsupportedConstraint{MOIU.variable_function_type(S), S}())
     end
-    return result
+    return Variable.concrete_bridge_type(b.variable_bridge_types[index], S)
 end
 function bridge_type(b::LazyBridgeOptimizer, F::Type{<:MOI.AbstractFunction}, S::Type{<:MOI.AbstractSet})
-    update!(b, (F, S))
-    result = get(b.constraint_best, (F, S), nothing)
-    if result === nothing
+    index = bridge_index(b, F, S)
+    if iszero(index)
         throw(MOI.UnsupportedConstraint{F, S}())
     end
-    return result
+    return Constraint.concrete_bridge_type(b.constraint_bridge_types[index], F, S)
 end
 function bridge_type(b::LazyBridgeOptimizer, F::Type{<:MOI.AbstractScalarFunction})
-    update!(b, (F,))
-    result = get(b.objective_best, (F,), nothing)
-    if result === nothing
+    index = bridge_index(b, F)
+    if iszero(index)
         throw(MOI.UnsupportedAttribute(MOI.ObjectiveFunction{F}()))
     end
-    return result
+    return Objective.concrete_bridge_type(b.objective_bridge_types[index], F)
 end
 
 _func_name(::Type{Constraint.ScalarFunctionizeBridge}) = "SingleVariable", "constraint"
 _func_name(::Type{Constraint.VectorFunctionizeBridge}) = "VectorOfVariables", "constraint"
 _func_name(::Type{Objective.FunctionizeBridge}) = "SingleVariable", "objective"
-function _functionize_bridge(bridge_types, target_type)
+function __functionize_bridge(bridge_types, target_type)
```

Comment:
Having the single-underscore and double-underscore methods is confusing. 

Are the instances of `__functionize_bridge` above meant to be the single-underscore methods?